lib/TikZImage.pm

```diff
@@ -166,49 +168,57 @@ sub draw {
 	close $fh;
 
 	my $ext = $self->ext;
-	my $tex_ext = $ext eq 'svg' && $self->svgMethod eq 'dvisvgm' ? 'dvi' : 'pdf';
-	my $latex_binary = WeBWorK::PG::IO::externalCommand($tex_ext eq 'dvi' ? 'latex' : 'pdflatex');
-
-	# Generate the pdf file.
-	system "cd " . $working_dir . " && $latex_binary image.tex > pdflatex.stdout 2> /dev/null";
-
-	if (-r "$working_dir/image.$tex_ext") {
-		chmod(0777, "$working_dir/image.$tex_ext");
```

Comment:
You dropped this in your new code.  This was the fix for the permissions error that @pstaabp was getting on OSX with the pdf2svg method.

Reply:
Do you think this is needed for both methods? It seems harmless to do it for both. Just an extra line of code though now that I'm splitting it up.